store/sqlstore/channel_store.go

```diff
@@ -2882,42 +2882,37 @@ func (s SqlChannelStore) AutocompleteInTeam(teamID, userID, term string, include
 
 // TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19334)
 func (s SqlChannelStore) AutocompleteInTeamForSearch(teamId string, userId string, term string, includeDeleted bool) (model.ChannelList, error) {
-	deleteFilter := "AND DeleteAt = 0"
+	query := s.getQueryBuilder().
+		Select("C.*").
+		From("Channels AS C").
+		Join("ChannelMembers AS CM ON CM.ChannelId = C.Id").
+		Where(sq.And{sq.Or{sq.Eq{"C.TeamId": teamId},
+			sq.And{sq.Eq{"C.TeamId": ""}, sq.Eq{"C.Type": "G"}}},
+			sq.Eq{"CM.UserId": userId}}).Limit(50)
+
 	if includeDeleted {
-		deleteFilter = ""
+		query = query.Where(sq.And{sq.Eq{"DeleteAt": string("0")}})
 	}
 
-	queryFormat := `
-		SELECT
-			C.*
-		FROM
-			Channels AS C
-		JOIN
-			ChannelMembers AS CM ON CM.ChannelId = C.Id
-		WHERE
-			(C.TeamId = :TeamId OR (C.TeamId = '' AND C.Type = :ChannelType))
-			AND CM.UserId = :UserId
-			` + deleteFilter + `
-			%v
-		LIMIT 50`
-
+	likeClause, likeTerm := s.buildLIKEClause(term, "Name, DisplayName, Purpose")
```

Comment:
The original objective was to rewrite this method as well to return an `sq` object to avoid all the `strings.ReplaceAll` business, but we can do that later.